circleapi.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	circleci "github.com/maplebed/go-circleci"
+)
+
+// pollCircleAPI will continue to poll the Circle API for this workflow's status
+// until it succeeds or fails. It waits until this is the only job left running
+// then declares the workflow finished and records success or failure same as
+// the `handleBuild` function
+func pollCircleAPI(traceID, teamName, apiHost, dataset string) error {
+	token, _ := os.LookupEnv("CIRCLE_API_TOKEN")
+	workflowID, _ := os.LookupEnv("CIRCLE_WORKFLOW_ID")
+	thisJobName, _ := os.LookupEnv("CIRCLE_JOB")
+	client := &circleci.Client{
+		Token: token,
+	}
+
+	wfJobs, err := getJobs(client, workflowID)
+	if err != nil {
+		return err
+	}
+
+	// ok, we have our jobs. Find ourself, watch the rest, see if any fail, wait
+	// till they're all done
+	done := make(chan struct{})
+	tic := time.NewTicker(5 * time.Second)
+	var failed bool
```

Comment:
maybe make the `done` channel hold this bool instead of setting it within the goroutine.

Reply:
would that be an odd overloading of "what is the status" and "should we stop polling"? I thought the pattern of using `done` channels was to make them strictly the signal that the thing had happened. Hmm. Looking around, I'm not sure where I got that idea. Mixing the two makes it feel more like the return value of the function. 

meh. I'm not sure it makes flow any more clear. How strong a suggestion is this?